### PR TITLE
Add minimum node version to engine field in package.json

### PR DESCRIPTION
**Pull request by rgranger**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/4787

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self-explanatory."

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ It's mighty as it:
 - is **unopinionated** so that you can customize it to your exact needs
 - has a **growing community** and is used by [Facebook](https://code.facebook.com/posts/879890885467584/improving-css-quality-at-facebook-and-beyond/), [GitHub](https://github.com/primer/stylelint-config-primer) and [WordPress](https://github.com/ntwb/stylelint-config-wordpress/)
 
+Minimum NodeJs version supported : v10.13.0
```
Comment (jeddy3) [inline README.md:22]: We can remove this line as using the `engine` field in the `package.json` will suffice. It's the standardised way of communicating a package's compatibility and will produce a warning on the CLI when `npm i` is used, e.g.:

```
npm WARN notsup Unsupported engine for stylelint@13.4.0: wanted: {"node":">=10.13.0"} (current: {"node":"8.17.0","npm":"6.13.4"})
```

**Comment by jeddy3** [comment]
@rgranger Thanks for making the change.

You'll need to run `npm run format` to resolve the lint warning.

It'll likely reorder the engine property in the `package.json` to be consistent with stylelint's other repositories.

**Comment by rgranger** [comment]
done

**Comment by jeddy3** [comment]
Changelog:

> - Fixed: specify minimum node version in `package.json`'s `engine` field ([#4790](https://github.com/stylelint/stylelint/pull/4790)).

**Pull request merged**